feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/viewmodel/SessionDetailViewModel.kt

```diff
@@ -107,11 +165,44 @@ class SessionDetailViewModel @AssistedInject constructor(
         descriptionTextExpandStateLiveData.value = TextExpandState.EXPANDED
     }
 
+    fun thumbsUp(session: Session) {
+        val now = incrementThumbsUpCountLiveData.value ?: 0
+        val incremented = minOf(now + 1, MAX_APPLY_COUNT)
+        incrementThumbsUpCountLiveData.value = incremented
+
+        viewModelScope.launch {
+            incrementThumbsUpCountEvent.send(session.id to incremented)
+        }
+    }
+
+    @Suppress("EXPERIMENTAL_API_USAGE")
+    private suspend fun setupIncrementThumbsUpEvent() {
+        return incrementThumbsUpCountEvent.asFlow()
+            .debounce(INCREMENT_DEBOUNCE_MILLIS)
+            .catch { e ->
+                // TODO: Implement
+                Timber.debug { "⭐ error $e" }
+            }
+            .collect { (sessionId, count) ->
+                sessionRepository.incrementThumbsUpCount(
+                    sessionId = sessionId,
+                    count = count
+                )
+                Timber.debug { "⭐ increment $count posted" }
+                incrementThumbsUpCountLiveData.value = 0
```

Comment:
If you add the totalThumbsUpCount and the incrementThumbsUpCount to display, please note that some logic such as scan is required.
Perhaps if you put 0 in this timing, if you start with 10 thumbs-up and hit thumbs-up 5 times, it will instantly become 20thumbs-up.
If you don't add them, it's OK.